addons/payment/controllers/portal.py

```diff
@@ -175,6 +175,19 @@ def payment_method(self, **kwargs):
         }
         return request.render('payment.payment_methods', rendering_context)
 
+    @http.route('/my/payment_method/archive', type='json', auth='user')
+    def archive_token(self, token_id):
```

Comment:
Even for a quick fix, a full docstring never hurts.
```suggestion
    def archive_token(self, token_id):
        """ Check that a user has write access on a token and archive the token if so.

        :param int token_id: The token to archive, as a `payment.token` id
        :return: None
        """
```